lisa/operating_system.py

```diff
@@ -2001,6 +2002,117 @@ def set_kill_user_processes(self) -> None:
         )
         self._node.tools[Service].restart_service("systemd-logind")
 
+    def _replace_default_entry(self, entry: str) -> None:
+        self._log.debug(f"set boot entry to: {entry}")
+
+        # Check if GRUB_DEFAULT already exists in the file
+        grep_result = self._node.execute(
+            "grep -q '^GRUB_DEFAULT=' /etc/default/grub",
+            sudo=True,
+            no_error_log=True,
+        )
+
+        # substitute if GRUB_DEFAULT exists, otherwise append it
+        if grep_result.exit_code == 0:
+            sed = self._node.tools[Sed]
+            sed.substitute(
+                regexp="GRUB_DEFAULT=.*",
+                replacement=f"GRUB_DEFAULT='{entry}'",
+                file="/etc/default/grub",
+                sudo=True,
+            )
+        else:
+            self._node.execute(
+                f"echo \"GRUB_DEFAULT='{entry}'\" >> /etc/default/grub",
+                sudo=True,
+                shell=True,
+                expected_exit_code=0,
+                expected_exit_code_failure_message="Failed to append GRUB_DEFAULT",
+            )
+
+        # output to log for troubleshooting
+        cat = self._node.tools[Cat]
+        cat.run("/etc/default/grub")
+
+    def replace_boot_kernel(self, kernel_version: str) -> None:
+        self._log.info(
+            f"Configuring Grub to boot into kernel version: {kernel_version}"
+        )
+
+        # Extract the actual kernel version from RPM package name
+        # Examples:
+        # kernel-lvbs-6.6.89-9.cm2.x86_64 -> 6.6.89-9.cm2
+        # kernel-6.6.89-9.azl3.x86_64 -> 6.6.89-9.azl3
+        extracted_version = kernel_version
+        rpm_version_pattern = re.compile(
+            r"^kernel-(?:[^-]+-)*(?P<version>\d+\.\d+\.\d+.*?)\.x86_64$"
+        )
```

Comment:
Is it possible that we're using an ARM kernel, in which case this pattern may not apply?